### ISSUE TITLE
fix(astro-parser): spread attributes containing TypeScript or expression-child siblings are misparsed (v5 mirror of #3824)

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3856]
- Markuplint version: dev branch HEAD (v5 RC)
- Parser lang: Astro (`@markuplint/astro-parser`)
- Node.js version: 24
- OS: any

## Describe the bug

`packages/@markuplint/astro-parser/src/parser.ts` on the dev (v5 RC) branch carries the identical Astro spread-attribute parsing bug already fixed for v4 in #3854 (Issue #3824).

`visitAttr()` on dev still routes spread attributes through the upstream `safeScriptParser` (espree-based) without a brace-aware pre-pass, so:

1. `{...{ command: 'close' } as any}` is split into a partial spread plus bogus `as` / `any}` attributes — espree does not understand TypeScript syntax and terminates at `as`.
2. `<div {...props}>{label}</div>` (or any element where a spread attribute is followed by an expression child) throws `SyntaxError: Invalid tag syntax: ...` — espree's cover-wrap (`$={...}`) extends past the spread's closing `}` into surrounding HTML, treating `{...props}>{label}` as a binary `>` expression and swallowing the next sibling.

`packages/@markuplint/parser-utils/src/script-parser.ts` on dev still uses the same `cover` strategy as v4, so the root cause is identical.

## Steps To Reproduce

On dev:

```sh
git checkout dev
```

Add a parser test such as:

```ts
parser.parse('<div {...props}>{label}</div>');
```

It throws `ParserError: SyntaxError: Invalid tag syntax: "<div {...props}>{label}</div>"`.

## The current behavior

```
<markuplint> error: The "as" attribute is disallowed (invalid-attr)
<markuplint> error: The "any}" attribute is disallowed (invalid-attr)
<markuplint> error: SyntaxError: Invalid tag syntax: "<div {...props}>{label}</div>" (parse-error)
```

## The expected behavior

- `{...{ command: 'close' } as any}` should be one spread attribute.
- Astro-valid spread patterns (static, dynamic tag, conditional, multiple, component wrappers with descendant expression children) should parse without error.

## Why this is a separate Issue from #3824

- #3824 was scoped to v4 (now fixed via #3854 merged 2026-05-10).
- The dev (v5 RC) branch and v4 are diverged and cannot be merged (per project policy), so v4 fixes do not propagate automatically.

## Suggested fix

Forward-port #3854's commits to dev:

- `fe76c34f0` `fix(astro-parser): preserve spread attributes containing TypeScript and expression-child siblings`
- `8f3100951` `docs(astro-parser): document the brace-aware spread attribute extractor`
- `e544db07c` `style(astro-parser): apply Prettier formatting`
- `469f25fa2` `style(astro-parser): drop unnecessary type assertion in findMatchingBrace`

The astro-parser source on v4 and dev is structurally identical for the affected paths, so the patch should apply with minimal conflict (lint stack on dev is oxlint/oxfmt instead of ESLint/Prettier — keep that in mind for the final formatting commits).

## Tests

The 27 tests added in #3854 (`spread-attr.spec.ts` 19 + `parser.spec.ts` `#3824` describe 8) provide the regression coverage; they need to apply on dev.
